### PR TITLE
Update 03-prepare-user-data.md

### PR DESCRIPTION
**Pull request by rishi-anand**
## Describe the Change

This PR updates User Data examples for edge-native clusters.

## Review Changes

💻 [Add Preview URL]()

🎫 [Jira Ticket]()

### CODE REVIEW

**Comment by spectro-prow** [comment]
Welcome @rishi-anand! It looks like this is your first PR to spectrocloud/librarium 🎉

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 1]
```diff
@@ -236,4 +285,4 @@ install:
 
 # Next Steps
 
-The last step of the EdgeForce workflow is to build the Edge artifact. Check out the [Build Images](/clusters/edge/edgeforge-workflow/build-images) guide to learn how to build your images with the Edge Installer Builder CLI and create an Edge artifact.
\ No newline at end of file
+The last step of the EdgeForce workflow is to build the Edge artifact. Check out the [Build Images](/clusters/edge/edgeforge-workflow/build-images) guide to learn how to build your images with the Edge Installer Builder CLI and create an Edge artifact.
```
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:288]: I don't see a change here.

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 2]
```diff
@@ -84,44 +158,19 @@ stylus:
   poweroff: true
 ```
 
-**Site** - supplied at the edge location through a bootable USB drive.
+**Site** - supplied at the edge location through a bootable USB drive. `projectName` if specified, will override project information specified in stage user-data's edge host registration token. Optionally, additional tags can also be specified. If edge site requires proxy for outbound connection then it must be provided in site user-data.
```
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:161]:
```suggestion
**Site** - supplied at the edge location through a bootable USB drive. If specified, the `projectName` value overrides project information specified in the user data's edge host registration token. Optionally, you can add tags `tags:city`, `tags:building`, and `tags:zip-code`.
```

**Review by ritawatson — APPROVED** [review]
Minor comments. @karl-cardenas-coding you might want to double-check my suggested changes. Otherwise, LGTM.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1261#" title="Author self-approved">rishi-anand</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1261#pullrequestreview-1402860135" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 3]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+        *****************************
+        *****************************
+        ------END CERTIFICATE------
+      - |
+        ------BEGIN CERTIFICATE------
+        *****************************
+        *****************************
+        ------END CERTIFICATE------
+
+# there is no password specified to default kairos user. To access edge host console, we will need to specify authorized keys or password.
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:126]:
```suggestion
# There is no password specified to the default kairos user. You must specify authorized keys or passwords to access the Edge host console. 
```

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 4]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+
+    # name of the device, this may also be referred to as the edge id or edge host id.  If no edge host name is specified
+    # one will be generated from the device serial number.  If stylus cannot the device serial number a random id will
+    # be used instead. In the case of hardware that does not have a serial number is highly recommended to specify the
+    # device name as a random name is not deterministic and may lead to a device being registered twice under different
+    # names.
+    name: edge-appliance-1
+
+    # Optional
+    # If edge host requires proxy to connect to palette instance or to pull images then do specify proxy information here
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:90]:
```suggestion
    # Optional
    # If the Edge host requires a proxy to connect to Palette or to pull images, then specify the proxy information in this section
```

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 5]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+    tags:
+      key1: value1
+      key2: value2
+      key3: value3
+
+    # name of the device, this may also be referred to as the edge id or edge host id.  If no edge host name is specified
+    # one will be generated from the device serial number.  If stylus cannot the device serial number a random id will
+    # be used instead. In the case of hardware that does not have a serial number is highly recommended to specify the
+    # device name as a random name is not deterministic and may lead to a device being registered twice under different
+    # names.
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:86]:
```suggestion
    # The device's name, may also be referred to as the Edge ID or Edge host ID.  If no Edge hostname is specified,
    # a hostname will be generated from the device serial number.  If the Edge Installer cannot identify the device serial number, then a random ID will
    # be generated and used instead. In the case of hardware that does not have a serial number, we recommended specifying a
    # random name, with minimal chances of being re-used by a different Edge host.
```
Language cleanup.

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 6]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+    paletteEndpoint: api.spectrocloud.com
+
+    # edgeHostToken is auto-registration token when edge host registers with the saas by itself when it boots up
+    # this token can be generated by browsing to Tenant Settings -> Registration Token.
+    # specify name and select project id in which edge host should register itself
+    edgeHostToken: aUAxxxxxxxxx0ChYCrO
+
+    # projectUid <Optional :need to provide if edge host registration token is not generated using project id>
+    projectUid: 12345677788
+    # tags which will be assigned to devices as labels
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:76]:
```suggestion
    # Tags which that will be assigned to the device as labels.
```

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 7]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+  site:
+    # host for hubble api to register device.
+    paletteEndpoint: api.spectrocloud.com
+
+    # edgeHostToken is auto-registration token when edge host registers with the saas by itself when it boots up
+    # this token can be generated by browsing to Tenant Settings -> Registration Token.
+    # specify name and select project id in which edge host should register itself
+    edgeHostToken: aUAxxxxxxxxx0ChYCrO
+
+    # projectUid <Optional :need to provide if edge host registration token is not generated using project id>
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:74]:
```suggestion
    # The Palette project ID the Edge host should pair with. This is an optional field if an edgeHostToken is used and the token was assigned to a project.
```

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 8]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
 
 You may encounter the following scenarios when creating an Edge Installer configuration user data file. Use these examples as a starting point to help you create user data configurations that fit your needs. 
 
+## Sample User Data
+```yaml
+stylus:
+  site:
+    # host for hubble api to register device.
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:66]:
```suggestion
    # The Palette API endpoint to use. The default value is api.spectrocloud.com.
```

File: content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md [hunk 9]
```diff
@@ -59,6 +59,80 @@ For your initial testing, your user data may include global settings and site-sp
...
+    paletteEndpoint: api.spectrocloud.com
+
+    # edgeHostToken is auto-registration token when edge host registers with the saas by itself when it boots up
+    # this token can be generated by browsing to Tenant Settings -> Registration Token.
+    # specify name and select project id in which edge host should register itself
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/04-edgeforge-workflow/03-prepare-user-data.md:71]:
```suggestion
    # The edgeHostToken is an auto-registration token to register the Edge host with Palette upon bootup.
    # This token can be generated by navigating to the Tenant Settings -> Registration Token.
    # Specify a name and select the project id with which the Edge host should register itself.
```

**Review by karl-cardenas-coding — APPROVED** [review]
Looks good @rishi-anand just some minor language cleanup suggestions.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1261#pullrequestreview-1402864498" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1261#" title="Author self-approved">rishi-anand</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1261#pullrequestreview-1402860135" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**